docs/operations/hook-sidecar.md

```diff
@@ -0,0 +1,162 @@
+# Hook Sidecar Container
+
+## Introduction
+
+In KubeVirt, a Hook Sidecar container is a sidecar container (a secondary container that runs along with the main
+application container within the same Pod) used to apply customizations before the Virtual Machine is initialized. This
+ability is provided since configurable elements in the VMI specification do not cover all of the [libvirt domain
+XML](https://libvirt.org/formatdomain.html) elements.
+
+The sidecar containers communicate with the main container over a socket with a gRPC protocol. There are two main
+sidecar hooks:
+
+1. `onDefineDomain`: This hook helps to customize libvirt's XML and return the new XML over gRPC for the VM creation.
+2. `preCloudInitIso`: This hook helps to customize the cloud-init configuration. It operates on and returns JSON
+   formatted cloud-init data.
+
+## Enabling Sidecar feature gate
+
+Sidecar feature gate can be enabled by following the steps mentioned in [Activating feature gates](..
```

Comment:
It would be helpful to add the exact capitalization/spelling for the feature gate in backtick quotes so the reader knows precisely what they are enabling, even if that is `Sidecar`.

Reply:
Sure. I have modified in the section title as well. PTAL.